src/ray/core_worker/fiber.h

```diff
@@ -155,7 +157,6 @@ class FiberState {
 
   void Join() {
     fiber_stopped_event_->Wait();
-    fiber_runner_thread_.detach();
```

Comment:
Currently we are relying on the behavior that `channel_.close();` and `fiber_stopped_event_->Wait();` can be called multiple times, which I checked is true. But this seems fragile and rely on the underlying behavior of these libraries. Can we just have our own `stopped_` and `joined_`  flags and early return if they are already called? Thoughts @hongchaodeng @rynewang

Reply:
Let's merge this and fix the issue first.
We are basically guaranteeing something that the library do not provide -- thread cancellation.
That's orthogonal to enhancing the capabilities.